### PR TITLE
Bugfix/v2/plugin context binding

### PR DESCRIPTION
**Pull request by chipbell4**
Fixing some plugin context binding issues that were brought up during testing. Also removing some old promise-resolving logic that we really don't need.

### CODE REVIEW

**Comment by nickgravelyn** [comment]
👍

I think there’s some example code in the plugin folder that you’ll want to update as well but this is going to be very good for plugin developers. 

One question this brings to mind is whether there is use in having both `setup` and `preload` given that both are called at the same point. It might actually be simpler to just have `setup` return the promise and synchronous plugins can just do their work and `return Promise.resolve()`. Not necessary but might simplify the plugin api a little bit.

**Comment by SmithAF** [comment]
I agree as well that having both setup and preloads seems a little redundant, but I could also see an argument for keeping them depending on how plugins use those functions typically

File: src/Application.js [hunk 1]
```diff
@@ -92,11 +92,10 @@ export class Application {
     // Also attempt to fetch over the iframe barrier for old container support
     this.container.fetch('playOptions', e => (this.playOptions.value = e.data));
 
-    Application._plugins.forEach(plugin => plugin.setup.call(this));
+    Application._plugins.forEach(plugin => plugin.setup(this));
+
+    const preloads = Application._plugins.map(plugin => plugin.preload(this));
```
Comment (SmithAF) [inline src/Application.js]: This could be streamlined a tiny bit
```js
Promise.all(Application._plugins.map(plugin => plugin.preload(this)))
```

**Comment by chipbell4** [comment]
Ah, yes! Excellent ideas! I'll do that:

- Refactor `setup` to return a promise (or undefined)
- Remove `preload` altogether, in favor of letting `setup` be async.
- Cleanup some of that `Promise.all` logic.

**Pull request merged**